src/xma/src/xmaapi/xmadecoder.cpp

```diff
@@ -272,6 +262,38 @@ xma_dec_session_create(XmaDecoderProperties *dec_props)
     dec_session->base.hw_session.private_do_not_use = (void*) priv1;
     dec_session->base.session_signature = (void*)(((uint64_t)priv1) | ((uint64_t)priv1->reserved));
 
+    int32_t num_execbo = g_xma_singleton->num_execbos;
+    priv1->kernel_execbos.reserve(num_execbo);
+    priv1->num_execbo_allocated = num_execbo;
+    for (int32_t d = 0; d < num_execbo; d++) {
+        uint32_t  bo_handle;
+        int       execBO_size = MAX_EXECBO_BUFF_SIZE;

```

Comment:
This feels like duplicated code from xmaadmin.cpp?

Reply:
change coming in different PR